Code/GraphMol/SmilesParse/SmilesParse.h

```diff
@@ -121,36 +145,67 @@ inline RWMol *SmartsToMol(
 
 RDKIT_SMILESPARSE_EXPORT Atom *SmartsToAtom(const std::string &sma);
 RDKIT_SMILESPARSE_EXPORT Bond *SmartsToBond(const std::string &sma);
+}  // namespace v1
+
+namespace v2 {
+namespace SmilesParse {
+using RDKit::SmilesParse::SmartsParserParams;
+using RDKit::SmilesParse::SmilesParseException;
+using RDKit::SmilesParse::SmilesParserParams;
+
+// c++ core guidelines F.60: prefer T* over T& when "no argument" is a valid
+// option
+inline std::unique_ptr<RDKit::RWMol> SmilesToMol(
+    const std::string &smi, const SmilesParserParams *params = nullptr) {
```

Comment:
We can remove the nullptr entirely

const SmilesParams &params = SmileParams()

Reply:
This also removed the allocation of the default params every time.

Reply:
If we keep the bare pointer we probably should prevent copying this struct.

Reply:
There's a C++ core guideline which, I think, applies to this and recommends using a nullptr:
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rf-ptr-ref
I can get rid of the allocation by just have a static local.

Reply:
I'm happy to go the other way and use a reference with default, but following the core guidelines doesn't strike me as a terrible thing

Reply:
I don’t think this is necessarily true for default arguments.

https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#f51-where-there-is-a-choice-prefer-default-arguments-over-overloading

Reply:
In the core guidelines.

Reply:
that's an argument

Reply:
I would argue it’s an argument with a valid default value which I think isn’t the same thing.

Reply:
Valid here being not null.

Reply:
Unless we’re talking about something different :)

Reply:
I’ll make my case a little stronger this evening.

Reply:
No need. I agree with you.

Reply:
Yay!  I won an argument on the internet…. My son said it’s the proudest he’s ever felt about me.

Reply:
I'm glad I could be a part of it!